### PR TITLE
Fix Helm Chart Build

### LINKED ISSUE #43: Helm Chart Build Fail
When running `helm template charts/node-ipam-controller --values charts/node-ipam-controller/values.yaml` error occur: `mapping values are not allowed in this context`.

### PR DESCRIPTION
**Pull request by mneverov**
This PR fixes helm template and add chart linting and testing.

Better to review by commits:
- first - fixes template comment (must be [template comment](https://helm.sh/docs/chart_best_practices/templates/#comments-yaml-comments-vs-template-comments), not yaml comment)
- second - introduces helm chart linting and testing with [ct](https://github.com/helm/chart-testing). It also adds new makefile targets to set up a local test env and to run the controller.
 
Closes #43

### CODE REVIEW

**Comment by mneverov** [comment]
@aojea ptal

File: charts/node-ipam-controller/Chart.yaml [hunk 1]
```diff
@@ -17,4 +17,4 @@ name: node-ipam-controller
 description: A Helm chart for node-ipam-controller
 type: application
 version: 0.0.2
-appVersion: 0.0.2
+appVersion: 0.0.1
```
Comment (aojea) [inline charts/node-ipam-controller/Chart.yaml:20]: is this version downgrade expected?

File: .github/workflows/test_charts.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
           check-latest: true
 
       - name: Set up chart-testing
-        uses: helm/chart-testing-action@v2.7.0
+        uses: helm/chart-testing-action@v2.6.1
```
Comment (aojea) [inline .github/workflows/test_charts.yml:39]: if there is a problem with the version better add a comment explaining that this has to be pinned because ...

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -8,10 +8,6 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 We have full documentation on how to get started contributing here:
 
-<!---
```
Comment (aojea) [inline CONTRIBUTING.md:11]: there are multiple changes in this commit that makes it hard to parses, it seems tha READM and Dockerfile changes can be split from this

File: charts/node-ipam-controller/Chart.yaml [hunk 4]
```diff
@@ -16,5 +16,5 @@ apiVersion: v2
 name: node-ipam-controller
 description: A Helm chart for node-ipam-controller
 type: application
-version: 0.0.1
-appVersion: 0.0.1
+version: 0.0.2
+appVersion: 0.0.2
```
Comment (aojea) [inline charts/node-ipam-controller/Chart.yaml]: this is downgraded later , any reason?

File: .github/workflows/test_charts.yml (hunk 2 above)
Comment (mneverov) [reply .github/workflows/test_charts.yml:39]: in the [example](https://github.com/helm/chart-testing-action?tab=readme-ov-file#example-workflow) they use `v2.7.0`, but the actual last version is `2.6.1`.

File: charts/node-ipam-controller/Chart.yaml (hunk 1 above)
Comment (mneverov) [reply charts/node-ipam-controller/Chart.yaml:20]: I bumped it initially as I expected changes in the go code, but then downgraded again because only auxiliary scripts and charts were changed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (mneverov) [reply CONTRIBUTING.md:11]: I prefer to do few larg PRs, rather then many small. Should have done in another commit.
Readme update replaces explanation how to run the controller by copying commands with `make run`. 
I changed binary name from `manager` to `node-ipam-controller` because `manager` does not convey any message what the binary manages.

Comment (aojea) [reply CONTRIBUTING.md:11]: then better squash all commits, if I go commit by commit I can see things moving in one direction and later be reverted

Comment (mneverov) [reply CONTRIBUTING.md:11]: Usually, I try to do meaningful small commits, then squash&merge after review.
I should have done this here too, i.e. changing binary name as a separate commit, bumping chart version, updating readme. Sorry for the confusion.

Comment (aojea) [reply CONTRIBUTING.md:11]: heh, no worries, I do the same, is just that these multiple commits are part of your development process, and it seems you found some wrong assumptions doing that ... that is not interesting for the reviewer, that expects the multiple commits to be reviewable independently

File: README.md [hunk 5]
```diff
@@ -36,40 +36,21 @@ make image-build
...
-registry.
+### Outside Cluster
 
-Install node-ipam-controller in the cluster via helm:
+The following command runs a kind cluster and install CRDs.
+NOTE: run the controller with leader election disabled (see `IPAM_ENABLE_LEADER_ELECTION`) or specify `POD_NAME` and
+`POD_NAMESPACE` environment variables that are used as leader election ID and namespace.
 
 ```sh
-helm install node-ipam-controller ./charts/node-ipam-controller --create-namespace --namespace nodeipam --set image.tag=local
```
Comment (aojea) [inline README.md:74]: I expect to leave some instructions for users to run their controller in production, this is very developer focused only with `makefiles`

**Comment by aojea** [comment]
LGTM module add some instructions to the README so an user is able to understand how to deploy the controller in its cluster without make commands

**Comment by aojea** [comment]
/lgtm
/approve

Thanks
